File: sdks/ruby/lib/zerogpu/types/response.rb
# frozen_string_literal: true

module Zerogpu
  module Types
    class Response < Internal::Types::Model
      field :id, -> { String }, optional: false, nullable: false
      field :object, -> { String }, optional: false, nullable: false
      field :created, -> { Integer }, optional: false, nullable: false
      field :model, -> { String }, optional: false, nullable: false
      field :output, -> { Internal::Types::Array[Zerogpu::Types::OutputMessage] }, optional: false, nullable: false
      field :usage, -> { Zerogpu::Types::TokenUsage }, optional: true, nullable: false
    end
  end
end

File: sdks/typescript/api/types/Response.ts
import type * as ZerogpuApi from "../index.js";

export interface Response {
    id: string;
    object: string;
    /** Unix timestamp when the response was created */
    created: number;
    model: string;
    output: ZerogpuApi.OutputMessage[];
    usage?: ZerogpuApi.TokenUsage | undefined;
}
